### PR TITLE
CPLP-790 Connector SD factory registration

### PR DESCRIPTION
**Pull request by wira-mms**

### CODE REVIEW

**Comment by evegufy** [comment]
@Phil91 Can you review this PR?

**Review by jjeroch — CHANGES_REQUESTED** [review]
Looks overall very good. I have only one comment regarding a inline comment for the hardcoded values

File: src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs [hunk 1]
```diff
@@ -57,7 +61,30 @@ public async Task<ConnectorViewModel> CreateConnectorAsync(ConnectorInputModel c
...
+        using var transaction = await _repository.BeginTransactionAsync();
+
+        try
+        {
+            createdConnector = await _repository.CreateConnectorAsync(connector).ConfigureAwait(false);
+            var bpn = (await _companyRepository.GetCompanyByIdAsync(connectorInputModel.Provider))!.BusinessPartnerNumber!;
+            var httpClient = new HttpClient();
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+            var requestModel = new ConnectorSdFactoryRequestModel(bpn, "DE", "DE", connectorInputModel.ConnectorUrl, "connector", bpn, bpn, "BPNL000000000000");
+            response = await httpClient.PostAsJsonAsync(_settings.SdFactoryUrl, requestModel);
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs:76]: Done

**Comment by Phil91** [comment]
> @Phil91 Can you review this PR?

I will

**Review by Phil91 — COMMENTED** [review]
@jjeroch i'd like to discuss the future usage of repositories in the solution. We currently do have two ways to use those: 

1. directly injecting and using the repository (e.g the usage of the [`IConnectorsRepository`](https://github.com/catenax-ng/product-portal-backend/blob/feature/CPLP-790-connector-sd-factory/src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs#:~:text=public%20ConnectorsBusinessLogic(-,IConnectorsRepository,-connectorsRepository%2C%20ICompanyRepository) )
2. injecting [`IPortalRepositories`](https://github.com/catenax-ng/product-portal-backend/blob/main/src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/IPortalRepositories.cs) and use the [`GetInstance`](https://github.com/catenax-ng/product-portal-backend/blob/main/src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalRepositories.cs#:~:text=public%20RepositoryType-,GetInstance,-%3CRepositoryType%3E) implementation.

depending of the result of the discussion above, we would need to talk about the implementation of the transaction.

who would be the best person(s) for this discussion?

**Comment by Phil91** [comment]
> > @jjeroch i'd like to discuss the future usage of repositories in the solution. We currently do have two ways to use those:
> > 
> > 1. directly injecting and using the repository (e.g the usage of the [`IConnectorsRepository`](https://github.com/catenax-ng/product-portal-backend/blob/feature/CPLP-790-connector-sd-factory/src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs#:~:text=public%20ConnectorsBusinessLogic(-,IConnectorsRepository,-connectorsRepository%2C%20ICompanyRepository))
> > 2. injecting [`IPortalRepositories`](https://github.com/catenax-ng/product-portal-backend/blob/main/src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/IPortalRepositories.cs) and use the [`GetInstance`](https://github.com/catenax-ng/product-portal-backend/blob/main/src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalRepositories.cs#:~:text=public%20RepositoryType-,GetInstance,-%3CRepositoryType%3E) implementation.
> > 
> > depending of the result of the discussion above, we would need to talk about the implementation of the transaction.
> > who would be the best person(s) for this discussion?
> 
> @Phil91 for a detailed discussion as well as open brainstorming, Norbert is ideal. Additionally please include Evelyn where possible (that topic can likely also result into a longer discussion, but it will be interesting for Evelyn to understand the ideas as well and see the pro and cons for the different approaches). As soon as the topic is further defined and a "preferred approach" is agreed, please also include Akshay and Majid to ensure that they understand the preferred approach Can you please keep me in loop, I will not be the right person to discuss this in detail neither I will be able to join all discussions, however if possible I like to stay in the loop since I am always interested in improvements

@jjeroch i will setup a meeting for further discussion with Norbert and Evelyn (you as optional as well). In my opinion i could approve this PR, we would then need an additional task to refactor the repository usage depending on which usage we decide to take. Whats your opinion on that?

**Comment by jjeroch** [comment]
> > > @jjeroch i'd like to discuss the future usage of repositories in the solution. We currently do have two ways to use those:
> > > 
> > > 1. directly injecting and using the repository (e.g the usage of the [`IConnectorsRepository`](https://github.com/catenax-ng/product-portal-backend/blob/feature/CPLP-790-connector-sd-factory/src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs#:~:text=public%20ConnectorsBusinessLogic(-,IConnectorsRepository,-connectorsRepository%2C%20ICompanyRepository))
> > > 2. injecting [`IPortalRepositories`](https://github.com/catenax-ng/product-portal-backend/blob/main/src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/IPortalRepositories.cs) and use the [`GetInstance`](https://github.com/catenax-ng/product-portal-backend/blob/main/src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalRepositories.cs#:~:text=public%20RepositoryType-,GetInstance,-%3CRepositoryType%3E) implementation.
> > > 
> > > depending of the result of the discussion above, we would need to talk about the implementation of the transaction.
> > > who would be the best person(s) for this discussion?
> > 
> > 
> > @Phil91 for a detailed discussion as well as open brainstorming, Norbert is ideal. Additionally please include Evelyn where possible (that topic can likely also result into a longer discussion, but it will be interesting for Evelyn to understand the ideas as well and see the pro and cons for the different approaches). As soon as the topic is further defined and a "preferred approach" is agreed, please also include Akshay and Majid to ensure that they understand the preferred approach Can you please keep me in loop, I will not be the right person to discuss this in detail neither I will be able to join all discussions, however if possible I like to stay in the loop since I am always interested in improvements
> 
> @jjeroch i will setup a meeting for further discussion with Norbert and Evelyn (you as optional as well). In my opinion i could approve this PR, we would then need an additional task to refactor the repository usage depending on which usage we decide to take. Whats your opinion on that?

@Phil91 fully agree

**Review by Phil91 — APPROVED** [review]

**Pull request merged**